openedx/features/course_experience/__init__.py

```diff
@@ -104,5 +104,13 @@ def course_home_url(course_key):
     Arguments:
         course_key (CourseKey): The course key for which the home url is being requested.
     """
+    from openedx.core.djangoapps.catalog.utils import get_course_data
+    course_key_str = '{}+{}'.format(course_key.org, course_key.course)
+    course_data = get_course_data(course_key_str, ['course_type'])
+    if course_data:
+        course_type = course_data.get('course_type')
+        if course_type == settings.EXEC_ED_COURSE_TYPE:
```

Comment:
I believe we could implement this in a way that doesn't require explicitly hardcoding "executive education" conditionals into this method; even if those conditionals are configurable, it still hardcodes `EXEC_ED_*` settings into the code, which is technically 2U-specific business logic. It's going to continue to be difficult in having solid boundaries between what's 2U/edX and what's Open edX...

I might recommend something like the following to make this more flexible/generic/not Exec Ed specific:

```python
def course_home_url(course_key):
    """
    Returns the course home page's URL for the current user.

    Arguments:
        course_key (CourseKey): The course key for which the home url is being requested.
    """
    # configuration setting, temporarily hardcoded here. long-term, this would live in the LMS remote config 
    COURSE_HOME_URL_OVERRIDES_BY_COURSE_TYPE_AND_PRODUCT_SOURCE = [{
        "course_type": "executive-education-2u",
        "product_source": "2u",
        "course_home_url_override": "https://getsmarter.com/account",
    }]

    # logic for course home url overrides, in an open-source friendly way
    # solely via configuration without any hardcoded reference to Exec Ed.
    if COURSE_HOME_URL_OVERRIDES_BY_COURSE_TYPE_AND_PRODUCT_SOURCE:
        from openedx.core.djangoapps.catalog.utils import get_course_data
        course_key_str = '{}+{}'.format(course_key.org, course_key.course)
        course_data = get_course_data(course_key_str, ['course_type', 'product_source'])
        if not course_data:
            pass

        course_type = course_data.get('course_type')
        product_source = course_data.get('product_source')
        product_source_slug = product_source.get('slug') if product_source else None

        for override in COURSE_HOME_URL_OVERRIDES_BY_COURSE_TYPE_AND_PRODUCT_SOURCE:
            override_course_type = override.get('course_type')
            override_product_source = override.get('product_source')
            if override_course_type == course_type and override_product_source == product_source_slug:
                return override.get('course_home_url_override')

    # default course home url
    from .url_helpers import get_learning_mfe_home_url
    return get_learning_mfe_home_url(course_key, url_fragment='home')
```

Reply:
@muhammad-ammar All of the approaches in this PR seem to be bandaging the flaw in the upstream integration where the course_host_url should have been populated in the database ... so that we can avoid this "settings based logic"
maybe you can add this url as part of https://github.com/openedx/edx-platform/pull/32271/files#r1199872320